src/linnia.js

```diff
@@ -35,10 +37,18 @@ class Linnia {
     this._users = _util.truffleHack(_users);
     this._records = _util.truffleHack(_records);
     this._permissions = _util.truffleHack(_permissions);
-    // set linniaContractUpgradeHubAddress address
-    // using user defined address
-    this._hubAddress = opt.linniaContractUpgradeHubAddress;
-    this._tokenAddress = opt.linniaTokenContractAddress;
+    // set Hub Address
+    if(opt && opt.linniaContractUpgradeHubAddress){
```

Comment:
This name is weird. Would it be a lot of work to change it to something shorter?

Reply:
I agree, terrible name. I'll change it